osu.Game/Scoring/Legacy/LegacyScoreEncoder.cs

```diff
@@ -95,8 +95,9 @@ private string replayStringContent
 
                     foreach (var f in score.Replay.Frames.OfType<IConvertibleReplayFrame>().Select(f => f.ToLegacy(beatmap)))
                     {
-                        replayData.Append(FormattableString.Invariant($"{(int)Math.Round(f.Time - lastF.Time)}|{f.MouseX ?? 0}|{f.MouseY ?? 0}|{(int)f.ButtonState},"));
+                        replayData.Append(FormattableString.Invariant($"{Math.Round(f.Time - lastF.Time)}|{f.MouseX ?? 0}|{f.MouseY ?? 0}|{(int)f.ButtonState},"));
```

Comment:
This cast is required. The output type must be an `int`.

Reply:
> This cast is required. The output type must be an `int`.

No.
See [here](https://docs.microsoft.com/en-us/dotnet/api/system.math.round?view=net-5.0#System_Math_Round_System_Double_)
`Rounds a double-precision floating-point value to the nearest integral value`

Reply:
Read: https://docs.microsoft.com/en-us/dotnet/api/system.math.round?view=net-5.0#rounding-and-precision

Reply:
The value is immediately formatted, and an integral number of type `double` are formatted without a decimal point.
Thus, the cast shouldn't make difference unless the time difference is at least 2^31 (should never happen).

Reply:
> Read: https://docs.microsoft.com/en-us/dotnet/api/system.math.round?view=net-5.0#rounding-and-precision
> 
> I know what `Math.Round()` does. But not all numbers are perfectly representable by a float. I'm not taking chances with float precision nonsense.
> This is required in order for stable to correctly parse, they _must_ be 100% integral units.

``` C#
Console.WriteLine("Classic Math.Round in CSharp");
Console.WriteLine(Math.Round(4.4)); // 4
Console.WriteLine(Math.Round(4.5)); // 4
Console.WriteLine(Math.Round(4.6)); // 5
Console.WriteLine(Math.Round(5.5)); // 6
```

As @ekrctb said, it returns a `Double` but it makes no difference when being formatted into a string, unless the value is greater than 2^31 .

Reply:
Okay, I've tested and confirmed this is the case.